File: development/palindrome_word/PalindromeWord.java
public class PalindromeWord {
    public static boolean IsPalindrome(String word) {
        int left = 0;
        int right = word.length() - 1;

        while (left < right) {
            if(word.charAt(left) != word.charAt(right)) return false;

            left++;
            right--;
        }

        return true;
    }

    public static void main(String[] args) {
        String word1 = "repaper";
        String word2 = "intelligence";

        System.out.printf("the word: %s is a palindrome: %b%n", word1, IsPalindrome(word1));
        System.out.printf("the word: %s is a palindrome: %b%n", word2, IsPalindrome(word2));
    }
}

File: development/palindrome_word/PalindromeWord.cs
using System;

bool IsPalindrome(string word)
{
    int left = 0;
    int right = word.Length - 1;

    while(left < right)
    {
        if(word[left] != word[right]) return false;

        left++;
        right--;
    }

    return true;
}

string word1 = "madam";
string word2 = "palindrome";
Console.WriteLine("the word: " + word1 + " is a palindrome: " + IsPalindrome(word1));
Console.WriteLine("the word: " + word2 + " is a palindrome: " + IsPalindrome(word2));